File: common-client-api-libs/src/main/java/eu/getsoftware/hotelico/clients/api/clients/infrastructure/menu/dto/MenuOrderDTO.java
package eu.getsoftware.hotelico.clients.api.clients.infrastructure.menu.dto;

import eu.getsoftware.hotelico.clients.common.dto.BasicDTO;
import eu.getsoftware.hotelico.clients.common.utils.ControllerUtils;
import lombok.Getter;
import lombok.Setter;
import lombok.With;
import lombok.experimental.SuperBuilder;

import java.sql.Timestamp;
import java.util.Date;
import java.util.List;

@With
@Getter
@SuperBuilder
public class MenuOrderDTO extends BasicDTO{
    long hotelId = 0L;
    long senderId = 0L;
    int itemAmount = 0;
    String orderLocation = "";
    String firstName = "";
    String lastName = "";
    String customerComment = "";
    
    @Setter
    String orderStatus = null;
    double totalPrice = 0.0;
    double totalMoney = 0;
    String orderCode = "";
    private String previewPictureUrl = null;
    Date validFrom = null;
    Date validTo = null;
    boolean orderInRoom = false;
    
    @Setter
    boolean closed = false;
    Timestamp timestamp = null;
    List<MenuItemDTO> menuItems;
    
    private String timeString = null;

    String getPreviewPictureUrl() {
        return previewPictureUrl!=null ? previewPictureUrl: ControllerUtils.PREVIEW_MENU_NOT_AVAILABLE_URL;
    }

    void setPreviewPictureUrl(String previewPictureUrl) {
        this.previewPictureUrl = previewPictureUrl;
    }
    
    String getTimeString() {
        return (timestamp != null)? ControllerUtils.getTimeFormatted(timestamp) : null;
    }

    void setTimeString(String timeString) {
        this.timeString = timeString;
    }
}

File: hotel-app/src/main/kotlin/eu/getsoftware/hotelico/menu/infrastructure/dto/MenuOrderDTO.kt
package eu.getsoftware.hotelico.menu.infrastructure.dto;

import eu.getsoftware.hotelico.clients.infrastructure.dto.BasicDTO
import eu.getsoftware.hotelico.clients.infrastructure.utils.ControllerUtils
import java.sql.Timestamp
import java.util.*

class MenuOrderDTO: BasicDTO
{
    var hotelId: Long? = 0L

    var senderId: Long? = 0L

    var itemAmount = 0

    var orderLocation = ""

    var firstName = ""

    var lastName = ""

    var customerComment = ""

    var orderStatus: String? = null

    var totalPrice = 0.0

    var totalMoney: Double? = null

    var orderCode = ""
    
    private var previewPictureUrl: String? = null

    var validFrom: Date? = null

    var validTo: Date? = null

    var orderInRoom = false

    var closed = false

    var timestamp: Timestamp? = null

    private var timeString: String? = null

    var menuItems = arrayOfNulls<MenuItemDTO>(0)

    constructor(): super()
    
    fun getPreviewPictureUrl(): String {
        return previewPictureUrl ?: ControllerUtils.PREVIEW_MENU_NOT_AVAILABLE_URL
    }

    fun setPreviewPictureUrl(previewPictureUrl: String) {
        this.previewPictureUrl = previewPictureUrl
    }
    
    fun getTimeString(): String? {
        return if (timestamp != null) ControllerUtils.getTimeFormatted(timestamp) else null
    }

    fun setTimeString(timeString: String) {
        this.timeString = timeString
    }
}
